Gem/Code/Source/Components/NetworkPlayerMovementComponent.cpp

```diff
@@ -124,6 +124,15 @@ namespace MultiplayerSample
 
     bool NetworkPlayerMovementComponentController::ShouldProcessInput() const
     {
+        // skip player input if the gameplay is in between a round (rest period)
+        if (const auto networkMatchComponent = AZ::Interface<NetworkMatchComponent>::Get())
+        {
+            if (networkMatchComponent->GetRoundTime() <= 0 && networkMatchComponent->GetRoundRestTimeRemaining() > 0)
```

Comment:
It would be better to move this entire logic inside Network Match Component and just create a call IsPlayerMovement/ActionAllowed().

Reply:
+1

Reply:
Or keep a set of player entities and call into them from the match enabling or disabling input processing.. Out of curiosity, does this cause corrections?

Reply:
Thanks! Cleaned up NetWeapon using the same logic as well.

Reply:
I haven't seen any corrections personally, but I would think it's possible. Towards the end of rounds, CreateInput on the player's side would think that the round is still happening and try to send input until it gets the message that the round is actually over